File: server/app/domain/locator.py
class Locator:
    def __init__(self, namespace, name):
        self.namespace = namespace
        self.name = name

    def getNamespace(self):
        return self.namespace

    def getName(self):
        return self.name

class LocatorSerializationService:
    @staticmethod
    def serialize(locator):
        return locator.namespace + "/" + locator.name

    @staticmethod
    def deserialize(string):
        slashIndex = string.find("/")
        if slashIndex != -1:
            namespace = string[0:slashIndex]
            name = string[slashIndex+1:]
            return Locator(namespace, name)
        else:
            raise DeserializationException("Error deserializing " + string)

class DeserializationException(Exception):
    pass

File: src/domain/locator.js
export class Locator {
  constructor(namespace, name) {
    this.namespace = namespace;
    this.name = name;
  }

  getNamespace() {
    return this.namespace;
  }

  getName() {
    return this.name;
  }

  toString() {
    return this.namespace + "/" + this.name;
  }
}

export class LocatorSerializationService {
  static serialize(locator) {
    return locator.toString();
  }

  static deserialize(string) {
    const slashIndex = string.indexOf("/");
    if (slashIndex !== -1) {
      const namespace = string.slice(0, slashIndex);
      const name = string.slice(slashIndex + 1);
      return new Locator(namespace, name);
    } else {
      throw new DeserializationException();
    }
  }
}

export class DeserializationException {}
